### DISCUSSION TITLE
can't transfer discussion

### DISCUSSION THREAD

**Discussion by Morriz** [community/community#3381] [Discussions] [answered]
When choosing to transfer a discussion, none of our repos is offered, and none match search. We have multiple open source repos, as well as multiple private ones. Can anybody transfer a discussion to another repo?

**Comment by PranamBhat** [comment]
Hi @Morriz 

Go to your repository:

Go to settings and make sure you have selected the **Discussions** checkbox:

![image](https://user-images.githubusercontent.com/32120033/108307765-6c5d6e80-71d4-11eb-80f4-0597db6a5a35.png)


Now go to "**Discussions**" page and click on "**Transfer discussion**". It will ask you to choose a repository. Make sure you have enabled **Discussion** under **Settings** for the destination repository as well.


You can check more here : [https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions-in-your-repository](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions-in-your-repository)


Have a nice day!

**Reply by Morriz** [reply]
I don't see that option in a repo, and I am org owner. I do see the option in org settings to turn on discussions for the entire organization though.

**Reply by PranamBhat** [reply]
Yes. True. That is the scenario for the **Organization**. **Enable**/turn it on and transfer your discussion.

**Reply by Morriz** [reply]
Obviously I have that turned on otherwise I wouldn't be asking for help to transfer a discussion ;)

So how do we do this?

**Reply by PranamBhat** [reply]
Go to "**Discussions**" and click on "**Transfer discussion**". It will ask you to choose a repository. Choose which repository you want to transfer. That's all.

**Reply by Morriz** [reply]
Did you not read my initial post? I said it does not offer any repos to transfer to.

**Comment by Shoeboxam** [comment]
~I recently ran into the same issue and noticed that you can only transfer discussions out of public repositories.~ There are similar rules for issues. I find it frustrating that, even as an admin, I can't do something that could just be a warning.

EDIT: Others have replicated the issue when attempting public to public transfers.

**Comment by tolot27** [comment]
@Shoeboxam Is this statement confirmed by GitHub? I try to transfer discussions, too but no repository is listed even if I own several repositories with discussions enabled.

**Reply by Shoeboxam** [reply]
I do not know if this behavior is confirmed by GitHub. It is the behavior I observe when I attempt to transfer issues and discussions.
- when the issue/discussion is public, I can transfer to any public or private repository in the org with said feature enabled
- when the issue/discussion is private, I can only transfer to private repositories in the org with said feature enabled

**Comment by curquiza** [comment]
I can't transfer discussion, from a public repo ([meilisearch](https://github.com/meilisearch/meilisearch)) to another public repo ([product](https://github.com/meilisearch/product/)). Both of them accept discussions

But I get this error for any try of transfer!




<img width="579" alt="Capture d’écran 2022-04-11 à 18 14 59" src="https://user-images.githubusercontent.com/20380692/162784820-3e9a24d0-b126-4da4-8a7e-6992c2f1068d.png">

<img width="539" alt="Capture d’écran 2022-04-11 à 18 15 03" src="https://user-images.githubusercontent.com/20380692/162784839-48cec4b5-f799-41b6-915c-73dac9cc34eb.png">

<img width="785" alt="Capture d’écran 2022-04-11 à 18 12 01" src="https://user-images.githubusercontent.com/20380692/162784675-ec059fac-6002-49b5-99f2-f2dcc570f786.png">

**Comment by brycewray** [comment]
Having the same issue as @curquiza mentioned. I am the owner of two public repos ([hugo_site](https://github.com/brycewray/hugo_site) and [astro-site](https://github.com/brycewray/astro-site)), both of which have Discussions enabled, and cannot transfer any discussions from one to the other (I use [giscus](https://giscus.app)).

cc: @laymonage

**Comment by brycewray** [comment]
**Does anybody with any power over GitHub Discussions read these?** Can we at least get an answer? Any information is better than just deafening silence, especially when it's clear that this problem has been going on for quite some months now (there were other reports from last year in a different Discussion). If it's no longer possible to transfer Discussions, then just remove the option and tell us. In the absence of new information, it's very difficult to depend on Discussions as any sort of reliable tool.

cc: @jayceeday @evi-liu

**Comment by evi-liu** [comment]
👋 Hi folks, the criteria for repositories you can transfer discussions to are the following:

- the target repository is under the same organization
- the target repository has Discussions enabled 
- the category type of the source discussion has a matching category type in the target repository

Are these criteria true for your transfers?

**Reply by brycewray** [reply]
@evi-liu Yes to all three in my case — same org., each repo has Discussions enabled, each repo has Announcement category, and each Discussion is in that category. Thanks for the reply!

**Reply by brycewray** [reply]
@evi-liu ... so does that help? Does the GH team know what’s causing this? Is there an ETA for a fix? Surely would appreciate some related info, even if it’s “We don’t know but it’s been worked on” or something similar.

**Reply by laymonage** [reply]
Actually, now that I think about it, does it work if you change the category to anything other than one that has the Announcements type, before transferring the discussion?

**Reply by brycewray** [reply]
@laymonage That worked! I changed each from Announcements to General, and transfer for each worked perfectly! Then, in the other repo, I changed each back to Announcements, and once again they appear in giscus. **Thank you, sir!**

**Reply by laymonage** [reply]
Great! I think I understand the issue, I've explained a bit more at https://github.com/github/feedback/discussions/3381#discussioncomment-2656034.

**Reply by curquiza** [reply]
Hello, I did not succeed to transfer a discussion categorized as "Open ended discussion" (name of the discussion: `General`) to a repo where only one "Open ended discussion" category exists, not called `General` but called `Feedback & Feature Proposal`
To fix the problem, I had to
- create a mirror category `Feedback & Feature Proposal` in the first repo
- change the category of the discussion from `General` to `Feedback & Feature Proposal`
- transfer the discussion in the second repo

Is it expected? I would like not to have a `Feedback & Feature Proposal` category in my first repo

**Comment by laymonage** [comment]
Okay, I think I understand the issue here. This issue happens if you're trying to transfer a discussion that belongs to a category that uses the "Announcements" discussion format (not necessarily named "Announcements"), e.g.

<img width="456" alt="image" src="https://user-images.githubusercontent.com/6379424/165794923-31209021-ba53-490a-8212-7a6815bb0deb.png">

It says,

> Only maintainers and admins can post new discussions in these categories

So, apparently GitHub's transfer mechanism doesn't count as "maintainers"/"admins", so it fails when it's transferring the discussion.

The transfer works if you change the discussion to another category that uses a different discussion format before transferring. After the transfer, you can move it back to the correct category just fine (confirmed at https://github.com/github/feedback/discussions/3381#discussioncomment-2655855).

I think this info would be very useful for your engineers @evi-liu!

**Reply by evi-liu** [reply]
Ah yes, this makes sense and is definitely a bug. Thank you so much for getting to the bottom of this and sharing!

**Reply by jsoref** [reply]
Filed https://github.com/github/docs/issues/31057 (and https://github.com/github/docs/issues/31056)

**Comment by King-of-Infinite-Space** [comment]
I'm still having problem transferring some discussions after trying all suggestions above. Directly created discussions can be transferred, but some discussions that were converted from issues cannot.

**Comment by jsoref** [comment, marked as answer]
@Morriz, can I encourage you to mark one of these posts as an answer? Prior to this comment, I'd suggest https://github.com/orgs/community/discussions/3381#discussioncomment-2656034 as it covers the most surprising aspect of your question and chains to the other half of the answer.

For everyone else, I've filed https://github.com/github/docs/issues/31057 about the lack of documentation about constraints for transferring discussions.

I believe the requirements are:
1. The destination repository must have discussions enabled.
2. The discussion being transferred cannot be an "Announcement format".
3. The destination repository _probably_ can't be more public than the source repository. See [visibility](https://docs.github.com/en/enterprise-cloud@latest/repositories/creating-and-managing-repositories/about-repositories#about-repository-visibility). I'm not absolutely certain about this, but at least for public/private, that should be correct.
4. There must be a matching category type in the destination repository.
5. You must have permissions to create discussions in the repository where you want to transfer the discussion.
6. _I'm not sure if you need specific permissions to the source repository._

**Reply by Morriz** [reply]
I trust your suggestion. It's been too long ago

**Reply by jsoref** [reply]
Feel free to mark my answer as the answer.

**Comment by rafalkrupinski** [comment]
I couldn't transfer a discussion between two public repos in the same org.

Turned out I had to open the Discussions tab and click `Get started` :D

**Comment by 54145a** [comment]
I can't transfer a discussion between two public repos from general to general, in the same org. It matches all the criteria for discussion transferring.
<img width="520" height="998" alt="image" src="https://github.com/user-attachments/assets/0a19c2c1-8299-4a3e-b614-d2a32511f8ab" />
<img width="1329" height="594" alt="image" src="https://github.com/user-attachments/assets/b8146dae-7b6b-4ee5-b6f3-59c0e3fa0688" />
